WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java

```diff
@@ -76,48 +75,77 @@ protected void onCreate(Bundle savedInstanceState) {
         if (isSharingText()) {
-            mActionSpinner.setVisibility(View.GONE);
+            mActionGroup.setVisibility(View.GONE);
             findViewById(R.id.action_spinner_title).setVisibility(View.GONE);
             // if text/plain and only one blog, then don't show this fragment, share it directly to a new post
             if (mSiteNames.length == 1) {
                 // Single site, startActivityAndFinish will pick the first one by default
                 startActivityAndFinish(new Intent(this, EditPostActivity.class));
             }
         } else {
-            String[] actions = new String[]{getString(R.string.share_action_post), getString(
-                    R.string.share_action_media)};
-            ArrayAdapter<String> actionAdapter = new ArrayAdapter<String>(this,
-                    R.layout.spinner_menu_dropdown_item, actions);
-            mActionSpinner.setAdapter(actionAdapter);
-            mActionSpinner.setOnItemSelectedListener(this);
+            mActionGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(RadioGroup group, @IdRes int checkedId) {
+                    mActionId = checkedId;
+                }
+            });
         }
         loadLastUsed();
     }
 
     @Override
-    public void onBackPressed() {
-        super.onBackPressed();
-        finish();
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        if (parent.getId() == R.id.blog_spinner) {
+            mSelectedSiteLocalId = mSiteIds[position];
+        }
     }
 
     @Override
     public void onNothingSelected(AdapterView<?> parent) {
-        // noop
+        // nop
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String permissions[],
+                                           @NonNull int[] grantResults) {
+        if (requestCode == SHARE_MEDIA_PERMISSION_REQUEST_CODE) {
+            for (int grantResult : grantResults) {
+                if (grantResult == PackageManager.PERMISSION_DENIED) {
+                    ToastUtils.showToast(this, getString(R.string.add_media_permission_required));
+                    return;
+                }
+            }
+            shareIt();
+        }
+    }
+
+    /**
+     * Callback for "Share" button.
+     */
+    public void onShareClicked(View view) {
+        shareIt();
+    }
+
+    /**
+     * Callback for "Cancel" button.
+     */
+    public void onCancelClicked(View view) {
+        finish();
     }
 
     private void finishIfNoVisibleBlogs() {
-        // If not logged in, then ask to log in, else inform the user to set at least one blog
-        // visible
+        // If not logged in, then ask to log in, else inform the user to set at least one blog visible
         if (FluxCUtils.isSignedInWPComOrHasWPOrgSite(mAccountStore, mSiteStore)) {
```

Comment:
This isn't a result of this PR, but this routine looks wrong to me. Shouldn't this be if **not** isSignedInWPComOrHasWPOrgSite?